src/main/scala/com/microsoft/hyperspace/actions/RefreshIncrementalAction.scala

```diff
@@ -56,46 +54,27 @@ class RefreshIncrementalAction(
       "Refresh index is updating index by removing index entries " +
         s"corresponding to ${deletedFiles.length} deleted source data files.")
 
-    if (appendedFiles.nonEmpty) {
+    lazy val dfWithAppendedFiles = {
```

Comment:
Because it seems like the code is independent of index implementations. A framework shouldn't delegate something that is not a concern of the implementation.

Please refer to the revised interface.